typescript/infra/scripts/get-owner-ica.ts

```diff
@@ -60,58 +128,319 @@ async function main() {
+  // The owner of the routing ISM is the ICA itself.
+  const desiredIsmConfig = getIcaIsm(
+    icaChain,
+    mainnet3Deployer,
+    icaArtifact.ica,
+  );
+  const ismModule = new EvmIsmModule(multiProvider, {
+    chain: icaChain,
+    config: desiredIsmConfig,
+    addresses: {
+      ...(chainAddresses[icaChain] as any),
+      deployedIsm: icaArtifact.ism,
+    },
+  });
+  const actualIsmConfig = await ismModule.read();
+
+  if (
+    !deepEquals(
+      normalizeConfig(actualIsmConfig),
+      normalizeConfig(desiredIsmConfig),
+    )
+  ) {
+    console.log('ISM mismatch for', icaChain);
+    console.log('actualIsmConfig:', JSON.stringify(actualIsmConfig));
+    console.log('desiredIsmConfig:', JSON.stringify(desiredIsmConfig));
+    return false;
+  }
+
+  const chainOwnerConfig = {
+    origin: originChain,
+    owner: originOwner,
+    ismOverride: icaArtifact.ism,
+  };
+
+  // Then, confirm that the ISM address is recoverable.
+  const account = await ica.getAccount(icaChain, chainOwnerConfig);
+
+  // Try to recover the account
+  if (eqAddress(account, icaArtifact.ica)) {
+    return true;
+  } else {
+    console.error(
+      `⚠️⚠️⚠️ Failed to recover ICA for ${icaChain}. Expected: ${
+        icaArtifact.ica
+      }, got: ${account}. Chain owner config: ${JSON.stringify(
+        chainOwnerConfig,
+      )} ⚠️⚠️⚠️`,
+    );
+    return false;
+  }
+}
+
+// -- ISM config generation --
+
+const merkleRoot = (multisig: MultisigConfig): MultisigIsmConfig => ({
+  type: IsmType.MERKLE_ROOT_MULTISIG,
+  ...multisig,
+});
+
+const messageIdIsm = (multisig: MultisigConfig): MultisigIsmConfig => ({
+  type: IsmType.MESSAGE_ID_MULTISIG,
+  ...multisig,
+});
+
+const aggregationIsm = (multisig: MultisigConfig): AggregationIsmConfig => ({
+  type: IsmType.AGGREGATION,
+  modules: [messageIdIsm(multisig), merkleRoot(multisig)],
+  threshold: 1,
+});
+
+function getIcaIsm(
+  originChain: string,
+  deployer: string,
+  routingIsmOwner: string,
+): IsmConfig {
+  const multisig = defaultMultisigConfigs[originChain];
+  const awValidator =
+    awValidators[originChain as keyof typeof awValidators].validators?.[0];
+  // Ensure the AW validator was found and is in the multisig.
+  if (
+    !awValidator ||
+    !multisig.validators.find((v) => eqAddress(v, awValidator))
+  ) {
+    throw new Error(
+      `AW validator for ${originChain} (address: ${awValidator}) found in the validator set`,
+    );
+  }
+
+  // A routing ISM so that the ISM is mutable without requiring a new ICA,
+  // as the ICA address depends on the ISM address.
+  return {
+    type: IsmType.ROUTING,
+    owner: routingIsmOwner,
+    domains: {
+      [originChain]: {
+        type: IsmType.AGGREGATION,
+        modules: [
+          // This will always use the default ISM.
+          // We burn ownership and have no domains in the routing table.
+          {
+            type: IsmType.FALLBACK_ROUTING,
+            owner: '0xdead00000000000000000000000000000000dead',
```

Comment:
Is this a typical burn address on ethereum?

Reply:
I think `0x000000000000000000000000000000000000dEaD` is more commonly used

Reply:
happy to move to that